lint/importShadow_checker.go

```diff
@@ -0,0 +1,61 @@
+package lint
+
+//! Detects when imported package names shadowed in assignments.
+//
+// @Before:
+// import (
+// 	   "fmt"
+//     "math"
+// )
+// func main() {
+//     fmt.Println(math.Pi)
+//     // shadowing math package
+//     math := 10
+//     fmt.Println(len)
+// }
+//
+// @After:
+// import (
+//     "fmt"
+//     "math"
+// )
+// func main() {
+//     fmt.Println(math.Pi)
+//     // change identificator name
+//     value := 10
+//     fmt.Println(value)
+// }
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/go-critic/go-critic/lint/internal/astwalk"
+)
+
+func init() {
+	addChecker(&importShadowChecker{}, attrSyntaxOnly)
+}
+
+type importShadowChecker struct {
+	checkerBase
+}
+
+func (c *importShadowChecker) Init() {}
```

Comment:
You may remove `Init` method as this checker has no state that needs to be initialized.